web3/_utils/method_formatters.py

```diff
@@ -505,9 +510,31 @@ def raise_block_not_found(params: Tuple[BlockIdentifier, bool]) -> NoReturn:
     raise BlockNotFound(f"Block with id: {block_identifier} not found.")
 
 
+def raise_block_not_found_for_uncle_at_index(
+    params: Tuple[BlockIdentifier, Union[HexStr, int]]
+) -> NoReturn:
+    block_identifier = params[0]
+    uncle_index = to_integer_if_hex(params[1])
```

Comment:
(Carrying on from the last comment) I do think this explicit pattern is easier to reason about than the `block_number_formatter`. What do you think about renaming `block_number_formatter` to `to_hex_if_integer`, since this formatter has bled outside of just block formatting? (Okay if not in this PR.)

Reply:
Yep, agreed. I had a duplicate method in the original PR, so I'll add that here, and then add an issue to remove the duplicated method in a separate PR.